lib/persistence.js

```diff
@@ -204,20 +204,24 @@ Persistence.prototype.persistNewState = function (newDocs, cb) {
  * machine understandable collection
  */
 Persistence.prototype.treatRawData = function (rawData) {
-  var data = rawData.split('\n')
+  var i, doc
+    , data = rawData.split('\n')
     , dataById = {}
     , tdata = []
-    , i
     , indexes = {}
     , corruptItems = -1   // Last line of every data file is usually blank so not really corrupt
     ;
 
   for (i = 0; i < data.length; i += 1) {
-    var doc;
-
     try {
       doc = model.deserialize(this.beforeDeserialization(data[i]));
       if (doc._id) {
+
+        // Check for expired documents to delete
+        if (doc.$$deleted !== true && model.isExpired(doc)) {
+          doc.$$deleted = true;
+        }
```

Comment:
Although this works well, I think I'm actually going to remove it as it prevents me from detecting the removal of expired documents.